packages/ui-causal-graph-editor/src/shared/rendering/edge/symbols.tsx

```diff
@@ -54,11 +54,17 @@ export function createSideSymbol(
         }
 
         if (style.editorMode === EditorMode.EDGE_ENCODER) {
+            // Edge encoder - shows semi-circle tip for soft directed constraint
+            if (style.constraint?.type === EdgeConstraintType.SOFT_DIRECTED) {
+                gfx.lineStyle({
```

Comment:
I think we want to add 'cap: round' like in other arrow types below here, this makes the ends of the arrow rounded rather than sharp